src/middleware/routeHandler.js

```diff
@@ -162,21 +179,46 @@ const attachRollbackRouteHandlerWrapper = (routeHandler) => async (
   await routeHandler(req, res, next).catch(async (err) => {
     try {
       if (shouldUseGitFileSystem) {
-        await backOff(() => {
-          const rollbackRes = gitFileSystemService
-            .rollback(siteName, originalCommitSha)
+        await backOff(async () => {
+          const rollbackRes = await gitFileSystemService
+            .rollback(siteName, originalStagingCommitSha, STAGING_BRANCH)
+            .andThen(() =>
+              gitFileSystemService.rollback(
+                siteName,
+                originalStagingLiteCommitSha,
+                STAGING_LITE_BRANCH
+              )
+            )
             .unwrapOr(false)
           if (!rollbackRes) throw new GitFileSystemError("Rollback failure")
         })
         await backOff(() => {
-          const pushRes = gitFileSystemService
-            .push(siteName, true)
-            .unwrapOr(false)
+          let pushRes = gitFileSystemService.push(
+            siteName,
+            STAGING_BRANCH,
+            true
+          )
+          if (originalStagingLiteCommitSha) {
+            pushRes = pushRes.push(siteName, STAGING_LITE_BRANCH, true)
+          }
+
+          pushRes = pushRes.unwrapOr(false)
           if (!pushRes) throw new GitFileSystemError("Push failure")
         })
       } else {
-        await backOff(() => {
-          revertCommit(originalCommitSha, siteName, accessToken)
+        await backOff(async () => {
+          await revertCommit(
+            originalStagingCommitSha,
+            siteName,
+            accessToken,
+            STAGING_BRANCH
+          )
+          await revertCommit(
+            originalStagingLiteCommitSha,
+            siteName,
+            accessToken,
+            STAGING_LITE_BRANCH
+          )
```

Comment:
Same for this step!

Reply:
2e06bec48fd39e96336769aa4ca9b5b70ac81d19